tests/test_cli_tools.py

```diff
@@ -225,3 +228,22 @@ def test_mark_production(run_dir_root: Path):
     assert link_path.is_dir()
     assert link_path.is_symlink()
     assert link_path.resolve() == run_dir
+
+
+@pytest.mark.parametrize(('last_stage_version', 'last_stage_directory', 'last_stage_root', 'result'),
+                         [(ABSOLUTE_PATH_1, ABSOLUTE_PATH_2, None, ABSOLUTE_PATH_2),
+                          (ABSOLUTE_PATH_1, ABSOLUTE_PATH_2, ABSOLUTE_PATH_1, ABSOLUTE_PATH_2),
+                          (ABSOLUTE_PATH_1, None, None, ABSOLUTE_PATH_1),
+                          (ABSOLUTE_PATH_1, None, ABSOLUTE_PATH_2, ABSOLUTE_PATH_1),
+                          (None, ABSOLUTE_PATH_1, None, ABSOLUTE_PATH_1),
+                          (ABSOLUTE_PATH_1, RELATIVE_PATH, None, None),
```

Comment:
It threw me a little when I saw this, because I thought "the result should be an Error, shouldn't it?"

Then I glanced down and read the test function.

Consider splitting the success and error scenarios into separate tests as in this case the parametrize arguments almost function as a sort of truth table